controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -123,37 +152,108 @@ func (w *Workload) ControlPlaneIsHealthy(ctx context.Context) (HealthCheckResult
 		name := node.Name
 		response[name] = nil
 
+		var owningMachine *clusterv1.Machine = nil
+		// If there is no owning Machine for the node, setting conditions are no-op.
+		for _, machine := range controlPlane.Machines {
+			if machine.Status.NodeRef == nil {
+				continue
+			}
+			if machine.Status.NodeRef.Name == node.Name {
+				owningMachine = machine
+			}
+		}
+
+		// If there is any problem with the checks below, set response to any one of the problems.
+		var anyError error = nil
```

Comment:
Can we use a slice of errors like we do everywhere else and then use NewAggregate?